src/json.hpp.re2c

```diff
@@ -132,15 +159,156 @@ using is_unscoped_enum =
 
 namespace detail
 {
-// very useful construct against boilerplate (more boilerplate needed than in
-// C++17: http://en.cppreference.com/w/cpp/types/void_t)
-template <typename...> struct make_void
+template <typename Json> std::string type_name(const  Json& j)
```

Comment:
I don't really like the name `Json` here, because it is so easy to confuse with `json`. I have not found a better name though.